examples/nemo/eg6/README.md

```diff
@@ -0,0 +1,98 @@
+# PSyclone NEMO Example 6 - Read-only Verification
+
+**Author:** J. Henrichs, Bureau of Meteorology
+
+This example demonstrates the use of PSyclone to add code that checks
+if variables that are only read are actually modified (e.g. because
+of memory overwrite).
+
+Once you have installed PSyclone, you can transform the file using:
+
+```sh
+psyclone -s ./read_only_check.py dummy.f90
+```
+
+Executing this will output the transformed Fortran code with the 
+read-only-verification code added. 
+
+The generic read-only verification library in
+``../../../lib/read_only/generic`` is used, and will also be
+automatically compiled if required.
+
+## Compiling and Execution
+
+To create and compile the example, type ``make compile``.
+This example can be compiled and executed. It will report nothing,
+since no read-only variable is overwritten. But you can verify that
+the variables are checked by setting ``PSYDATA_VERBOSE=2``:
+
+```sh
+$ PSYDATA_VERBOSE=2 ./dummy 
+ PSyData: PreStart dummy r0
+ PSyData: DeclareScalarChar: dummy r0: char_var
+ PSyData: DeclareScalarLogical: dummy r0: logical_var
+ PSyData: DeclareScalarChar: dummy r0: char_var
+ PSyData: DeclareScalarLogical: dummy r0: logical_var
+ PSyData: checked variable char_var
+ PSyData: checked variable logical_var
+ PSyData: PostEnd dummy r0
+   3.00000000     F
+```
+
+If you copy the lines 32 and 33 from ``dummy.f90`` into ``psy.f90``,
+the code will modify ``logical_var`` (by using out-of-bound array accesses.
+Or you could just manually set ``logical_var = .true.``). If you then
+compile again, an error will be produced.
```

Comment:
This worked for me. Good stuff. Only wrinkle is that doing `make compile` again ran PSyclone and over-wrote my modified psy.f90 file. I had to cut-n-paste the compile and link lines to build with the modified psy.f90 file. Not a biggie. Either mention it here or perhaps it's just a small tweak to the Makefile?

Reply:
Good point. I think I've fixed that now.